buffer/src/test/java/io/netty/buffer/AbstractByteBufTest.java

```diff
@@ -2766,6 +2767,22 @@ public void testSetBytesAfterRelease3() {
     }
 
     @Test(expected = IllegalReferenceCountException.class)
+    public void testSetCharSequenceAfterRelease() {
+        testSetCharSequenceAfterRelease0(CharsetUtil.US_ASCII);
+        testSetCharSequenceAfterRelease0(CharsetUtil.ISO_8859_1);
+        testSetCharSequenceAfterRelease0(CharsetUtil.UTF_8);
+    }
+
+    private void testSetCharSequenceAfterRelease0(Charset charset) {
+        ByteBuf buffer = buffer();
+        try {
+            releasedBuffer().setCharSequence(0, "Test", charset);
+        } finally {
+            buffer.release();
```

Comment:
Why need second release? I think such will be enough:
```java
@Test(expected = IllegalReferenceCountException.class)
public void testSetCharSequenceAfterReleaseUtf8() {
    releasedBuffer().setCharSequence(0, "Test", CharsetUtil.UTF_8);
}

@Test(expected = IllegalReferenceCountException.class)
public void testSetCharSequenceAfterReleaseAscii() {
    releasedBuffer().setCharSequence(0, "Test", CharsetUtil.US_ASCII);
}
```
This approach is used in other tests for checking `IllegalReferenceCountException`.